sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/InExpressionBenchmark.scala

```diff
@@ -0,0 +1,260 @@
+    val df = spark.range(1, numRows).select($"id".cast(TimestampType))
+    runBenchmark(name, df, values, numRows, minNumIters)
+  }
+
+  private def runDateBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Unit = {
+    val name = s"$numItems dates"
+    val values = (1 to numItems).map(n => 1970 + n).map(y => s"CAST('$y-01-01' AS date)")
+    val df = spark.range(1, numRows).select($"id".cast(TimestampType).cast(DateType))
+    runBenchmark(name, df, values, numRows, minNumIters)
+  }
+
+  private def runArrayBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Unit = {
+    val name = s"$numItems arrays"
+    val values = (1 to numItems).map(i => s"array($i)")
+    val df = spark.range(1, numRows).select(array($"id").as("id"))
+    runBenchmark(name, df, values, numRows, minNumIters)
+  }
+
+  private def runStructBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Unit = {
+    val name = s"$numItems structs"
+    val values = (1 to numItems).map(i => s"struct($i)")
+    val df = spark.range(1, numRows).select(struct($"id".as("col1")).as("id"))
+    runBenchmark(name, df, values, numRows, minNumIters)
+  }
+
+  private def runBenchmark(
+      name: String,
+      df: DataFrame,
+      values: Seq[Any],
+      numRows: Long,
+      minNumIters: Int): Unit = {
+
+    val benchmark = new Benchmark(name, numRows, minNumIters, output = output)
+
+    df.createOrReplaceTempView("t")
+
+    def testClosure(): Unit = {
+      val df = spark.sql(s"SELECT * FROM t WHERE id IN (${values.mkString(",")})")
+      df.queryExecution.toRdd.foreach(_ => Unit)
+    }
+
+    benchmark.addCase("In expression") { _ =>
+      withSQLConf(SQLConf.OPTIMIZER_INSET_CONVERSION_THRESHOLD.key -> values.size.toString) {
+        testClosure()
+      }
+    }
+
+    benchmark.addCase("InSet expression") { _ =>
+      withSQLConf(SQLConf.OPTIMIZER_INSET_CONVERSION_THRESHOLD.key -> "1") {
+        testClosure()
+      }
+    }
+
+    benchmark.run()
+  }
+
+  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
+    val largeNumRows = 10000000
+    val smallNumRows = 1000000
+    val minNumIters = 5
+
+    runBenchmark("In Expression Benchmark") {
+      runByteBenchmark(numItems = 5, largeNumRows, minNumIters)
+      runByteBenchmark(numItems = 10, largeNumRows, minNumIters)
+      runByteBenchmark(numItems = 25, largeNumRows, minNumIters)
+      runByteBenchmark(numItems = 50, largeNumRows, minNumIters)
+      runByteBenchmark(numItems = 100, largeNumRows, minNumIters)
+
+      runShortBenchmark(numItems = 5, largeNumRows, minNumIters)
+      runShortBenchmark(numItems = 10, largeNumRows, minNumIters)
+      runShortBenchmark(numItems = 25, largeNumRows, minNumIters)
+      runShortBenchmark(numItems = 50, largeNumRows, minNumIters)
+      runShortBenchmark(numItems = 100, largeNumRows, minNumIters)
+      runShortBenchmark(numItems = 200, largeNumRows, minNumIters)
```

Comment:
And, please use shorter form.
```scala
Seq(5, 10, 25, 50, 100, 200).foreach { numItems =>
  runShortBenchmark(numItems, largeNumRows, minNumIters)
}
```

Reply:
What about this?

```
  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
    val largeNumRows = 10000000
    val smallNumRows = 1000000
    val minNumIters = 5

    runBenchmark("In Expression Benchmark") {
      Seq(5, 10, 25, 50, 100, 200).foreach { numItems =>
        runByteBenchmark(numItems, largeNumRows, minNumIters)
        runShortBenchmark(numItems, largeNumRows, minNumIters)
        runIntBenchmark(numItems, largeNumRows, minNumIters)
        runLongBenchmark(numItems, largeNumRows, minNumIters)
        runFloatBenchmark(numItems, largeNumRows, minNumIters)
        runDoubleBenchmark(numItems, largeNumRows, minNumIters)
        runSmallDecimalBenchmark(numItems, smallNumRows, minNumIters)
        runLargeDecimalBenchmark(numItems, smallNumRows, minNumIters)
        runStringBenchmark(numItems, smallNumRows, minNumIters)
        runTimestampBenchmark(numItems, largeNumRows, minNumIters)
        runDateBenchmark(numItems, largeNumRows, minNumIters)
        runArrayBenchmark(numItems, smallNumRows, minNumIters)
        runStructBenchmark(numItems, smallNumRows, minNumIters)
      }
    }
  }
```

Reply:
It'll be difficult to see the result. You need to go back and forward to see the single type trends.